libs/stats/odc/stats/proc.py

```diff
@@ -166,6 +166,16 @@ def _safe_result(self, f: Future, task: Task) -> TaskResult:
             _log.error(f"Error during processing of {task.location} {e}")
             return TaskResult(task, error=str(e))
 
+    def _register_heartbeat(self, hearbeat_filepath: str):
+        """
+        Records the timestamp at which a hearbeat was detected
+
+        """
+        t_now = self.now()
+        file_obj = open(f"{hearbeat_filepath}/hearbeat.tmp", "w")
+        file_obj.write(t_now.strftime("%Y-%m-%d %H:%M:%S"))
+        file_obj.close()
```

Comment:
```suggestion
        with open(f"{hearbeat_filepath}/hearbeat.tmp", "w") as file_obj:
            file_obj.write(t_now.strftime("%Y-%m-%d %H:%M:%S"))
```

Reply:
Hold on, we want actual filepath supplied, not directory for it.

Reply:
Initially I was thinking of having a file with a known path that I can read in the liveness prob section of the job template.  We can use a variable inside the template and pass it to the cli as well as using it in the liveness check I suppose.

Reply:
yes, share path via ENV var between launch command and liveness probe script.